### DISCUSSION TITLE
Differing Behaviour on Testnet than Local Environment

### DISCUSSION THREAD

**Discussion by hooperben** [reach-sh/reach-lang#1085] [Q&A] [answered]
Hello!

When I run ./reach run on my local - everything runs fine.

The reach codebase is here:
https://github.com/hooperb/reach-hackathon/tree/main/packages/reach

However, when I deploy it to testnet and try to do this same behavior through a react app, I get a 400 error when I try to call the BluePayer.acceptGame() or RedPlayer.acceptGame() method. 

You can get the same error I'm getting by: 
1) going to https://algofour.com
2) connect with a wallet on testnet, then hit 'deploy contract'
3) Once the contract has been deployed, hit 'Join Game', and go to your wallet to confirm the transaction
4) Open the console, and you'll see a 400 request to the testnet-apo.algonode.cloud/v2/transaction, with a logged error of:
![image](https://user-images.githubusercontent.com/99215416/171504012-501d41ff-265c-4af1-855a-a8f60d936804.png)
**Error Message:**
`Error: BluePlayer.acceptGame errored with Error: API call failed: {
  "type": "signAndPost",
  "e": {
    "original": null,
    "response": {
      "message": "TransactionPool.Remember: transaction SG63MDCSKTTVB352DNPEOGCZTDDPCNAOZNZDH52L7AVN2HXHKU2Q: logic eval error: assert failed pc=1195. Details: pc=1195, opcodes=load 253\n<\nassert\n"
    },
    "status": 400
  },
  "es": "Error: Network request error. Received status 400: TransactionPool.Remember: transaction SG63MDCSKTTVB352DNPEOGCZTDDPCNAOZNZDH52L7AVN2HXHKU2Q: logic eval error: assert failed pc=1195. Details: pc=1195, opcodes=load 253\n<\nassert\n"
}`.

I can't make sense of this, as this is the exact same flow I'm using in the index.mjs, which functions correctly. Flow meaning:
- construct contract
- call acceptGame()

The contract is also definitely deploying correctly, as this variable in the logs:
![image](https://user-images.githubusercontent.com/99215416/171503911-e47d8406-e92f-4770-bebc-543506c596cf.png)

is created by the parallelReduce() and accessed by a view, so it is successfully launching.

Any help would be fantastic. Thank you

**Comment by jeapostrophe** [comment, marked as answer]
My guess is that your timeout is too small. It works in the devnet because we can pause the network while you're waiting to do work, but in the real world, things are happening behind the scenes.

**Reply by hooperben** [reply]
Yeah that was it, thanks mate.
